is.sh

```diff
@@ -1,132 +1,229 @@
-        number)
-            echo "$value_a" | grep -E '^[0-9]+(\.[0-9]+)?$'; return $?;;
-        older)
-            [ "$value_a" -ot "$value_b" ]; return $?;;
-        newer)
-            [ "$value_a" -nt "$value_b" ]; return $?;;
-        gt)
-            is not a number "$value_a"      && return 1;
-            is not a number "$value_b"      && return 1;
-            awk "BEGIN {exit $value_a > $value_b ? 0 : 1}"; return $?;;
-        lt)
-            is not a number "$value_a"      && return 1;
-            is not a number "$value_b"      && return 1;
-            awk "BEGIN {exit $value_a < $value_b ? 0 : 1}"; return $?;;
-        ge)
-            is not a number "$value_a"      && return 1;
-            is not a number "$value_b"      && return 1;
-            awk "BEGIN {exit $value_a >= $value_b ? 0 : 1}"; return $?;;
-        le)
-            is not a number "$value_a"      && return 1;
-            is not a number "$value_b"      && return 1;
-            awk "BEGIN {exit $value_a <= $value_b ? 0 : 1}"; return $?;;
-        eq|equal)
-            [ "$value_a" = "$value_b" ]     && return 0;
-            is not a number "$value_a"      && return 1;
-            is not a number "$value_b"      && return 1;
-            awk "BEGIN {exit $value_a == $value_b ? 0 : 1}"; return $?;;
-        match|matching)
-            echo "$value_b" | grep -xE "$value_a"; return $?;;
-        substr|substring)
-            echo "$value_b" | grep -F "$value_a"; return $?;;
-        true)
-            [ "$value_a" == true ] || [ "$value_a" == 0 ]; return $?;;
-        false)
-            [ "$value_a" != true ] && [ "$value_a" != 0 ]; return $?;;
-    esac > /dev/null
-
-    return 1
+    return 0
+  }
+
+  is::unset() {
+    command unset ${BASH_VERSION:+-f} is::unset \
+      is::show.version \
+      is::show.help \
+      is::tolower
+  }
+
+  [ "$#" -eq 0 ] && is::show.version && is::show.help && is::unset && return 0
+  [ "$1" = '--help' ] && is::show.help && is::unset && return 0
+  [ "$1" = '--version' ] && is::show.version && is::unset && return 0
+
+  # Since we currently support bash v3.2.57, we can't use `"${1,,}"`
+  local condition=$1 && shift 1
+  condition=$(is::tolower "$condition")
+
+  while [ "$condition" == 'a' ] || [ "$condition" == 'an' ] \
+    || [ "$condition" == 'the' ]; do :
+    condition=$(is::tolower "$1") && shift 1
+  done
+
+  if [ "$condition" = 'not' ]; then
+    ! is "${@}"
+    return $?
+  fi
+
+  # Note: case statements takes an expression & therefore doesn't need quotes
+  #       it is being kept for consistency
+  case "$condition" in
+    file)
+      [ -f "$1" ];;
+    dir|directory)
+      [ -d "$1" ];;
+    link|symlink)
+      [ -L "$1" ];;
+    existent|existing|exist|exists)
+      [ -e "$1" ];;
+    readable)
+      [ -r "$1" ];;
+    writeable)
+      [ -w "$1" ];;
+    executable)
+      [ -x "$1" ];;
+    available|installed)
+      # shellcheck disable=SC2230
+      which "$1";;
+    cmd|command)
```

Comment:
This condition is not documented in README.

Reply:
good catch!